### PR TITLE
feat: manifest.in + setup.cfg for pep517 compatible

### PR DESCRIPTION
**Pull request by aarnphm**
Signed-off-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

## Description

makes it pep517 compatible for setup.cfg

## Checklist:

- [x] My code follows the bentoml code style, both `make format` and
  `make lint` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [x] My change reduces project test coverage and requires unit tests to be added
- [x] I have added unit tests covering my code change
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -16,17 +14,9 @@ exclude MANIFEST.in
...
 prune docs
 prune .github
-
-# Patterns to exclude from any directory
-global-exclude *~
-global-exclude *.pyc
-global-exclude *.pyo
-global-exclude .ipynb_checkpoints
-global-exclude __pycache__
-global-exclude .DS_Store
```
Comment (parano) [inline MANIFEST.in:32]: I made a new build with your PR and see lots of .DS_Store files included in the distribution

File: MANIFEST.in [hunk 2]
```diff
@@ -1,10 +1,8 @@
 # All files tracked by Git are included in PyPI distribution
 
 # Files to exclude in PyPI distribution
-exclude .gitignore
-exclude .readthedocs.yml
-exclude .pep8speaks.yml
-exclude codecov.yml
+exclude .git*
+exclude *.yml
```
[the remark this answers is not in the document]
Comment (aarnphm) [reply MANIFEST.in:5]: noted

File: MANIFEST.in (hunk 1 above)
Comment (aarnphm) [reply MANIFEST.in:32]: i will add it back

**Comment by aarnphm** [comment]
close for now

**Pull request closed without merging**